include/tapkee/chain_interface.hpp

```diff
@@ -181,7 +184,10 @@ template <class DistanceCallback, class FeaturesCallback> class DistanceAndFeatu
      *        pointed by the first and the second arguments.
      */
     template <class KernelCallback>
-    CallbacksInitializedState<KernelCallback, DistanceCallback, FeaturesCallback> withKernel(
+        requires requires(KernelCallback callback, ScalarType a, ScalarType b) { callback.kernel(a, b); } ||
+                 requires(KernelCallback callback, const std::string& a, const std::string& b) { callback.kernel(a, b); } ||
```

Comment:
This hopefully won't have to be specialized in this way, I don't think it even helped adding the strings one.